[WIP] chore: add 3.12.0 english changelog

yutingzhao1991 opened the pull request:
- [x] english
- [x] chinese
- [x] wait feature branch ready

ndbroadbent committed:
Added middle and small sizes for Card component

orzyyyy committed:
feat: Add Input.Password

orzyyyy committed:
test: update snapshots

orzyyyy committed:
test: update snapshots

orzyyyy committed:
test: update snapshots

orzyyyy committed:
pref: fix Codacy problem

orzyyyy committed:
pref: fix Codacy problem

orzyyyy committed:
pref: fix codecov problem

zombieJ committed:
add eye-invisible icon

orzyyyy committed:
pref: add Icon type to Password => eye-invisible

caolvchong committed:
Modal.destroy():destroy all cofirm modal dialogs

caolvchong committed:
change Modal.destroy to Modal.destroyAll

zombieJ committed:
update snapshot

amedora committed:
add new color definitions

amedora committed:
use color definitions insted of numbers

orzyyyy committed:
pref: fix Codacy problem

orzyyyy committed:
pref: fix codecov problem

orzyyyy committed:
pref: fix codecov problem

orzyyyy committed:
style: remove indent

orzyyyy committed:
feat: add visibilityToggle to control whether Icon show

orzyyyy committed:
test: update test case

zombieJ committed:
ConfigProvider support prefixCls (#13389)

Basic support prefixCls.

orzyyyy committed:
test: update snapshots

zombieJ committed:
add forceRender support
fix #12503

afc163 committed:
:lipstick: site: tweak site style

chenshuai2144 committed:
Merge branch 'master' into feature

chenshuai2144 committed:
merge support prettier

zombieJ committed:
Merge pull request #13515 from ant-design/prettier

merge master

nick-ChenZe committed:
feat: add switcherIcon prop to Tree

nick-ChenZe committed:
feat: update docs for switcherIcon

chenshuai2144 committed:
merge master

chenshuai2144 committed:
Revert "Revert "Use less @plugin" (#13613)" (#13621)

This reverts commit 2c79d2a7c1a908fe9ba0b62e5202a3fd686f15b4.

chenshuai2144 committed:
Revert "Revert "Revert "Use less @plugin" (#13613)" (#13621)" (#13622)

This reverts commit f47aa91c5e0aa10bac9504b727f0e4ce6036173e.

chenshuai2144 committed:
Merge branch 'master' into feature

afc163 committed:
Merge branch 'master' into feature

zombieJ committed:
add missing less var

zombieJ committed:
update rc-cascader deps

minesaner committed:
Update use-with-create-react-app md

Add antd-react-scripts link

zombieJ committed:
update snapshot

zombieJ committed:
set form id on demo
fix #10218

JeffreyATW committed:
Fix primary color usage in .button-variant-other

Instead of using @primary-5 and @primary-7, use colorPalette mixin to modify @btn-primary-bg

zombieJ committed:
mention add defaultSuggestion

zombieJ committed:
merge origin/master

zombieJ committed:
update snapshot

zombieJ committed:
add test case

zombieJ committed:
update doc

zombieJ committed:
formItem support error message with reactNode (#13046)

close #13014

afc163 committed:
Merge branch 'master' into feature
* master:
  :memo: fix demo order
  Fixes #10576 (#13703)
  doc: update form & layout docs (#13701)
  update doc close #11802
  :bug: Fix Badge offset not working when count is ReactNode

afc163 committed:
Merge branch "master" into "feature"

afc163 committed:
:lipstick: Fix one tslint problem

afc163 committed:
✅ update test snapshot

aashutoshrathi committed:
Added Hindi Locale

afc163 committed:
Merge branch 'master' into feature

* master:
  feat(table): support table border-radius define in theme
  add form-tutorial.md
  :memo: Put gitter flags in badges in README (#13744)
  update deps of rc-select
  skip show-tooltip.md test when is dist test
  :lipstick: update slider demo
  Fix a regular expression
  Improvements to badges in README (#13743)

afc163 committed:
Merge master into feature branch

afc163 committed:
:up: upgrade rc-dropdown

afc163 committed:
:white_check_mark: fix snapshot

afc163 committed:
:tada: Dropdown.Button support `href` prop

zombieJ committed:
add options.preserve of form

rahgurung committed:
#13576 - Kannada locale added (kn_IN)

rahgurung committed:
added snapshot

afc163 committed:
:hammer: Merge master into feature

afc163 committed:
Merge branch 'feature' of github.com:ant-design/ant-design into feature

elfman committed:
Table: add prop `sortDirections` for table and column (#13773)

* Table: add prop `sortMethods`

with `sortMethods` user can change table sort methods and its order

* Table: `sortMethods` can be set on column prop

* rename `sortMethods` to `sortDirections`

afc163 committed:
:hammer: Merge master into feature

afc163 committed:
Merge branch 'master' into feature

afc163 committed:
:lipstick: fix lgtm alert

afc163 committed:
Merge branch 'master' into feature

zombieJ committed:
update snapshot

afc163 committed:
:hammer: Merge master into feature

zombieJ committed:
update modal demo

mraiguo committed:
Input.Group with TimePicker disappear icon: https://codepen.io/mraiguo/pen/QzvvvE?editors=0010

elfman committed:
DatePicker: show extra footer in all mode

afc163 committed:
Merge branch 'master' into feature

* master:
  Correct doc
  :up: ant-design-palettes => @ant-design/colors

afc163 committed:
:zap: renderFooter should applied to all mode, close #11618

afc163 committed:
:lipstick: fix dropdown button warning

elfman committed:
WeekPicker: support `renderExtraFooter`

elfman committed:
fix doc error and add test case for WeekPicker

afc163 committed:
:up: upgrade rc-time-picker

- add `popupStyle`
- add `onAmPmChange`

afc163 committed:
:up: upgrade rc-tree-select, no change need to mentioned

afc163 committed:
Merge branch 'master' into feature
* master:
  :memo: Add bodyStyle and style of Drawer in documentation, #13850
  :bug: Fix Drawer style not working
  :bug: fix markStyle affset style
  Update Santa.jsx
  update changelog
  bump 3.11.6
  revert Christmas egg
  [type] make type of components compatible with ComponentType<P>
  bump 3.11.5
  :sparkles: site: make document side bar sticky

afc163 committed:
:white_check_mark: fix snapshot

zombieJ committed:
merge master

zombieJ committed:
New Component: Empty (#13651)

afc163 committed:
Merge branch 'master' into feature
* master:
  Fixed Modal width when screen size smaller than 768px
  :rotating_light: coverage precision to 1
  :rotating_light: update .codecov.yml
  :lipstick: Fix Badge animation jump when count is ReactNode
  fix Popover arrow narrow line at top
  fix Popover arrow border in different directions
  fix popover arrow top shadow
  Change render of + & - icon fix #13924
  :bug: loose button type propTypes for custom usage
  fix input has min size close #3699
  fix: fix-picture-spell-error

afc163 committed:
:zap: Add `icon` to Modal.confirm/info/warning/error

deperated `iconType`

close #13918

afc163 committed:
:zap: support allowClear for Input

afc163 committed:
fix defaultValue

rahgurung committed:
added snapshot and updated package.json

rahgurung committed:
further fixes

afc163 committed:
:white_check_mark: fix snapshot and defaultValue warning

afc163 committed:
Merge branch 'master' into feature
* master: (23 commits)
  :white_check_mark: fix snapshots
  Add docs for the List rowKey prop
  :lipstick: don't reset font-family in each component
  :lipstick: chore card less variable
  Remove classNames from custom-typing file
  add collapsed faq ref: #13953
  feat: support define card border radius
  Fix dateRender not supported at WeekPicker
  :memo: Add links in Home Page
  :memo: Fix site responsive interaction, close #13954
  :lipstick: chore Steps[labelPlacement="vertical"] style
  :white_check_mark: fix warning in test case
  :memo: update tech antfin cases
  :bug: site menu open bug
  :bug: fix site defaultOpenKeys
  Correct docs
  :lipstick: chore button code style
  fix spaces in curly brackets
  fix two more code style issues
  Fix method argument name
  ...

afc163 committed:
:bug: Fix review problems

afc163 committed:
:white_check_mark: fix snapshots

afc163 committed:
:white_check_mark: fix onChange e.target.value when clear

zombieJ committed:
rate support 'tooltips'
close #13973

afc163 committed:
:bug: Fix review problems

afc163 committed:
:bug: Fix review problems

afc163 committed:
Fix input defaultValue and value

afc163 committed:
:memo: Add doc about Input[allowClear]

zombieJ committed:
Dropdown overlay support function callback (#14003)

* Dropdown's overlay prop support function callback.
* Dropdown 的 overlay 属性支持函数调用。

afc163 committed:
:bug: Fix Table filters trigger sort

close #13891
close #13933
close #13790

afc163 committed:
:bug: Fix Table custom title is not reachable

close #13984

yutingzhao1991 committed:
chore: add 3.12.0 english changelog

yutingzhao1991 committed:
chore: add 3.12.0 english changelog

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+- 🐞 Fixed arrow of Popover shadow. [#13935](https://github.com/ant-design/ant-design/pull/13935) [@crazyurus](https://github.com/crazyurus)
+- Badge
+  - 🐞 Fixed animation jump when count is ReactNode. [#13800](https://github.com/ant-design/ant-design/issues/13800)
+  - 🐞 Fixed `getScrollNumberTitle` typo bug. [#14005](https://github.com/ant-design/ant-design/pull/14005) [@nmsn](https://github.com/nmsn)
```
这个对用户无影响的可以去掉

Aladdin-ADD replied:
是否有必要规范化 commit message, 强制使用 tag: fix, feat, docs, etc.

这样就可以用脚本自动生成changelog

zy410419243 replied:
> 是否有必要规范化 commit message, 强制使用 tag: fix, feat, docs, etc.
> 
> 这样就可以用脚本自动生成changelog

不存在的，本来有 PR 就很开心了，总不能什么都指着路人吧...

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+- 🐞 Fixed Drawer prop `style` not working. [#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 🌟 Add new less var `upload-picture-card-border-style`. [#13919](https://github.com/ant-design/ant-design/pull/13919) [@flexchen](https://github.com/flexchen)
+  - 🐞 Fixed less var `upload-picture-card-size` typo. [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
```
这两条可以合并写，信息量聚焦一下

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+- 🌟 ConfigProvider support prefixCls. [#13389](https://github.com/ant-design/ant-design/pull/13389)
+- Less
+  - 🌟 Add more less variables for support custom colors definitions of some components. [#13425](https://github.com/ant-design/ant-design/pull/13425) [@amedora](https://github.com/amedora)
+  - 🐞 Fixed less typo of palette. [#13941](https://github.com/ant-design/ant-design/pull/13941) [@taoweicn](https://github.com/taoweicn)
```
对用户无影响

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+- Modal
+  - 🌟 Add `forceRender` support. [f791a50](https://github.com/ant-design/ant-design/commit/f791a50084eda5692635fd373bc2ec4a6b8d98e1)
+  - 🌟 Add `destroyAll`. [#13409](https://github.com/ant-design/ant-design/pull/13409) [@caolvchong](https://github.com/caolvchong)
+  - 🌟 Add `icon` to Modal.confirm/info/warning/error. [5c26635](https://github.com/ant-design/ant-design/commit/5c266355ec84d54d054ba89d35cc9280aef50a6e)
```
这里要点明废弃的 API

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+  - 🌟 Dropdown.Button support `href` prop. [607d080](https://github.com/ant-design/ant-design/commit/607d08094d4c59416c17d49e1ed5e87a166f61f7)
+  - 🌟 Add `openClassName`. [c6f267d](https://github.com/ant-design/ant-design/commit/c6f267d740d82ffc3e4f2f2a84cd3b2cc159c595)
+  - 🌟 Dropdown overlay support function callback. [#14003](https://github.com/ant-design/ant-design/pull/14003)
+  - 🐞 Fixed Dropdown `onVisibleChange` type define bug. [#13988](https://github.com/ant-design/ant-design/pull/13988) [@travikk](https://github.com/travikk)
```
ts 的归一类写吧

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+  - 🐞 Fixed Dropdown `onVisibleChange` type define bug. [#13988](https://github.com/ant-design/ant-design/pull/13988) [@travikk](https://github.com/travikk)
+- Button
+  - 💄 Update color with `.button-variant-other` className. [#13777](https://github.com/ant-design/ant-design/pull/13777) [@JeffreyATW](https://github.com/JeffreyATW)
+  - 💄 Migrate Button to new lifecycle methods. [#12752](https://github.com/ant-design/ant-design/pull/12752) [@natanielcz](https://github.com/natanielcz)
```
对用户无影响

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+  - 🌟 Add `icon` to Modal.confirm/info/warning/error. [5c26635](https://github.com/ant-design/ant-design/commit/5c266355ec84d54d054ba89d35cc9280aef50a6e)
+- Card
+  - 🌟 Added `middle` and `small` classes. [#13180](https://github.com/ant-design/ant-design/pull/13180) [@ndbroadbent](https://github.com/ndbroadbent)
+  - 🐞 Fixed less var `card-radius` bug. [#13966](https://github.com/ant-design/ant-design/pull/13966) [@vthinkxie](https://github.com/vthinkxie)
```
这里应该无变化，其实就是提上去了

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+  - 🌟 Dropdown overlay support function callback. [#14003](https://github.com/ant-design/ant-design/pull/14003)
+  - 🐞 Fixed Dropdown `onVisibleChange` type define bug. [#13988](https://github.com/ant-design/ant-design/pull/13988) [@travikk](https://github.com/travikk)
+- Button
+  - 💄 Update color with `.button-variant-other` className. [#13777](https://github.com/ant-design/ant-design/pull/13777) [@JeffreyATW](https://github.com/JeffreyATW)
```
对用户无影响

afc163 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,62 @@ timeline: true
+  - 🐞 Fixed dateRender not supported at WeekPicker. [#13957](https://github.com/ant-design/ant-design/issues/13957)
+  - 🐞 Fixed type define of DatePicker prop `disabledDate`. [#14008](https://github.com/ant-design/ant-design/pull/14008) [@vnguyen94](https://github.com/vnguyen94)
+- 🌟 Calendar support prop `renderExtraFooter` in all mode. [#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman)
+- 🌟 TimePicker support new props `popupStyle` and `onAmPmChange`. [833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662)
```
new prop

yutingzhao1991 committed:
fix review issue

yutingzhao1991 committed:
fix review issue

yutingzhao1991 committed:
add chinese changelog

yutingzhao1991 committed:
add chinese changelog

zy410419243 commented on CHANGELOG.zh-CN.md:
```diff
@@ -17,7 +17,53 @@ timeline: true
+- 🐞 修复 Drawer 的属性 `style` 不生效的问题。[#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 💄 添加新的 Less 变量 `upload-picture-card-border-style` 并修复 `upload-picture-card-size` 的拼写错误。[#13919](https://github.com/ant-design/ant-design/pull/13919) [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
+  - 🐞 修复在 Upload 组件中无法识别 `dpg` 后缀文件为图片的问题。[#13988]
```
~13988 没贴地址~

呃，是指向错了..https://github.com/ant-design/ant-design/pull/14013

zy410419243 commented on CHANGELOG.en-US.md:
```diff
@@ -15,6 +15,56 @@ timeline: true
+- 🐞 Fixed Drawer prop `style` not working. [#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 💄 Add new less var `upload-picture-card-border-style` and Fixed `upload-picture-card-size` typo. [#13919](https://github.com/ant-design/ant-design/pull/13919) [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
+  - 🐞 Fixed recognize `dpg` file type as image in Upload. [#13988]
```
这里指向错了，https://github.com/ant-design/ant-design/pull/14013

zombieJ committed:
Support collaborators PR template (#14023)

* add collaborator tmpl

* update tmpl

* update read me

* fix markdown style

* update readme

afc163 committed:
:up: upgrade devDependency

afc163 committed:
:bug: fix progress text align issue (#14024)

* :bug: fix progress text align issue

close #13997

* :white_check_mark: Set thresholds for codecov build failures

orzyyyy committed:
pref: add missing drag event of Tree (#14032)

close #14031

afc163 committed:
:lipstick: Drawer close icon should scroll with content (#14030)

* :lipstick: Drawer close icon should scroll with content

like its title

close #13811

* :white_check_mark: fix snapshots

zombieJ committed:
update snapshot (#14042)

afc163 committed:
:memo: Add instruction for one column without width left in fixed table

zombieJ committed:
merge master

zombieJ committed:
adjust table fixed column z-index (#14036)

fix #13930

afc163 committed:
Merge pull request #14046 from ant-design/snapshot

Feature merge master

afc163 committed:
:bug: Fix steps style under IE9

close #14001

ztplz committed:
Fix Affix flickering when scrolling

ztplz committed:
update

zombieJ committed:
Update rc-calendar to use new lifecycle methods & input auto focus (#14051)

close #11848

zombieJ committed:
Add full PR template link in template (#14061)

* add full tmpl link

* adjust

* simplify it

* add cn link

zombieJ committed:
ConfigConsumer support `withConsumer` (#14062)

* add withConfigConsumer

* clean up

gyh9457 committed:
fix: top border disappeared under some ie9

afc163 committed:
:bug: tweak style for not affecting vertical steps

meihuanyu committed:
fix calendar month range display (#14049)

ztplz committed:
Fix tslint warning

ztplz committed:
Correct typescript usage (#14074)

afc163 committed:
:memo: enhance upload documentation

afc163 committed:
:bug: Fix disabled button style in DatePicker panel

https://user-images.githubusercontent.com/507615/50693143-17505400-1071-11e9-9114-b150bef8f7f6.png

afc163 committed:
:memo: site: fix BackTop been covered by footer

close #14093

afc163 committed:
Update PULL_REQUEST_TEMPLATE.md

afc163 committed:
Update pr_cn.md

ztplz committed:
Update table docs

afc163 committed:
:bug: Fix nested Timeline last item missing line (#14110)

close #14108

zombieJ committed:
init Spin should also support delay trigger (#14105)

fix #14100

Go7hic committed:
bugfix

ztplz committed:
Format

picodoth committed:
Chore/fix master conflict (#14126)

* adjust table fixed column z-index (#14036)

fix #13930

* :memo: Add instruction for one column without width left in fixed table

* Add full PR template link in template (#14061)

* add full tmpl link

* adjust

* simplify it

* add cn link

* Fix Affix flickering when scrolling

* update

* fix: top border disappeared under some ie9

* fix calendar month range display (#14049)

* Fix tslint warning

* Correct typescript usage (#14074)

* :bug: Fix steps style under IE9

close #14001

* :bug: tweak style for not affecting vertical steps

* :memo: enhance upload documentation

* Input.Group with TimePicker disappear icon: https://codepen.io/mraiguo/pen/QzvvvE?editors=0010

* :bug: Fix disabled button style in DatePicker panel

https://user-images.githubusercontent.com/507615/50693143-17505400-1071-11e9-9114-b150bef8f7f6.png

* :memo: site: fix BackTop been covered by footer

close #14093

* Update PULL_REQUEST_TEMPLATE.md

* Update pr_cn.md

* :bug: Fix nested Timeline last item missing line (#14110)

close #14108

* init Spin should also support delay trigger (#14105)

fix #14100

* Update table docs

* bugfix

* Format

* update snapshot

* test: fix spin test

picodoth committed:
Merge branch 'feature' into master

picodoth committed:
Merge branch 'chore/3.12.0' of https://github.com/ant-design/ant-design into chore/3.12.0

picodoth committed:
docs: update 3.12.0 changelog

picodoth commented:
<img width="842" alt="screen shot 2019-01-06 at 3 15 44 pm" src="https://user-images.githubusercontent.com/1731837/50733289-fc432880-11c5-11e9-8e40-243d540aa7d0.png">
如上图 diff 抽风了，重新搞一个 pr: https://github.com/ant-design/ant-design/pull/14128
